test/rules_test.js

```diff
@@ -205,6 +205,14 @@ describe('rules', () => {
         ['2x + 7y + 5 + 3y + 9x + 11', '(2 x + 9 x) + (7 y + 3 y) + (5 + 11)'],
     ])
 
+    suite('add polynomials', rules.ADD_POLYNOMIAL_TERMS, [
+        ['2x + 2x + 2 + 4', '4 x + (2 + 4)'],
+        ['2x + 2y', '2 x + 2 y'],
+        ['2x + 3x + 2y + 3y', '5 x + 5 y'],
+        ['-2y + 2x - 3x^2', '-2y + 2x - 3x^2'],
+        ['-2y + 3y', '1 y'],
```

Comment:
You may want to add some cases that include terms with multiple variables factors or factors with exponents and definitely some with subtraction.